x-pack/plugins/maps/server/routes.js

```diff
@@ -417,44 +411,52 @@ export function initRoutes(router, licenseUid, mapConfig, kbnVersion, logger) {
         .replace('{y}', request.query.y)
         .replace('{z}', request.query.z);
 
-      return await proxyResource({ url }, { ok, badRequest });
+      return await proxyResource({ url }, response);
     }
   );
 
   router.get(
     {
       path: `${ROOT}/${EMS_TILES_API_PATH}/${EMS_GLYPHS_PATH}/{fontstack}/{range}`,
-      validate: false,
+      validate: {
+        params: schema.object({
+          fontstack: schema.string(),
+          range: schema.string(),
+        }),
+      },
     },
-    async (context, request, { ok, badRequest }) => {
+    async (context, request, response) => {
       if (!checkEMSProxyEnabled()) {
-        return badRequest('map.proxyElasticMapsServiceInMaps disabled');
+        return response.badRequest('map.proxyElasticMapsServiceInMaps disabled');
       }
       const url = mapConfig.emsFontLibraryUrl
         .replace('{fontstack}', request.params.fontstack)
         .replace('{range}', request.params.range);
 
-      return await proxyResource({ url }, { ok, badRequest });
+      return await proxyResource({ url }, response);
     }
   );
 
   router.get(
     {
       path: `${ROOT}/${EMS_TILES_API_PATH}/${EMS_SPRITES_PATH}/{id}/sprite{scaling?}.{extension}`,
       validate: {
+        query: schema.object({
+          elastic_tile_service_tos: schema.maybe(schema.string()),
+          my_app_name: schema.maybe(schema.string()),
+          my_app_version: schema.maybe(schema.string()),
+          license: schema.maybe(schema.string()),
```

Comment:
nit, but up to you really. we repeat some of these schemas such as `elastic_tile_service_tos: schema.maybe(schema.string()),` several times in this file. would it be feasible and easy to avoid duplicating these schemas?

Reply:
Yeah, I know what you mean, it does get a little repetitive. As far as I can tell, there's not an easy way to create reusable validation that we can also build upon (i.e.- add more validation where needed) unless we use a [custom validator](https://github.com/elastic/kibana/blob/9318862f1967d6fefdfca5e94417ff1617e5ba06/packages/kbn-config-schema/README.md#L501) that does some base validation such as checking `elastic_tile_service_tos: schema.maybe(schema.string())` etc. before doing any additional validation required by the calling route. tbh I'm still not sure that's the right path either so kicking that can down the road for now to get this fix in.